From d6f0bc89ae3d01d4a4425b4fee635e993aeb8c9a Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Sun, 5 Feb 2017 11:02:54 -0500
Subject: [PATCH] termcolor: add support for output to standard error

This is essentially a rename of the existing `Stdout` type to `StandardStream`
and a change of its constructor from a single `new()` function to have two
`stdout()` and `stderr()` functions.

Under the hood, we add add internal IoStandardStream{,Lock} enums that allow
us to abstract between Stdout and Stderr conveniently. The rest of the needed
changes then fall out fairly naturally.

Fixes #324.

[breaking-change]
---
 src/args.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/args.rs b/src/args.rs
index eeccb18..1973a9b 100644
--- a/src/args.rs
+++ b/src/args.rs
@@ -182,8 +182,8 @@ impl Args {
     }
 
     /// Create a new writer for single-threaded searching with color support.
-    pub fn stdout(&self) -> termcolor::Stdout {
-        termcolor::Stdout::new(self.color_choice)
+    pub fn stdout(&self) -> termcolor::StandardStream {
+        termcolor::StandardStream::stdout(self.color_choice)
     }
 
     /// Returns a handle to stdout for filtering search.
-- 
2.11.1

